stetho/src/main/java/com/facebook/stetho/inspector/network/DefaultAsyncPrettyPrinterRegistry.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+package com.facebook.stetho.inspector.network;
+
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.GuardedBy;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class DefaultAsyncPrettyPrinterRegistry implements AsyncPrettyPrinterRegistry {
+
+  @GuardedBy("this")
+  private final Map<String, AsyncPrettyPrinterFactory> mRegistry = new HashMap<>();
+
+  private static DefaultAsyncPrettyPrinterRegistry sInstance;
+
+  public static synchronized DefaultAsyncPrettyPrinterRegistry getInstance() {
+    if (sInstance == null) {
+      sInstance = new DefaultAsyncPrettyPrinterRegistry();
+    }
+    return sInstance;
+  }
+
+  @Override
+  public synchronized boolean register(String headerName, AsyncPrettyPrinterFactory factory) {
+    if (mRegistry.containsKey(headerName)) {
```

Comment:
Inefficient and a bad pattern.  If a caller wishes to register twice, you should either allow it or throw as a warning.  I prefer simply allowing it.

FWIW, `put` returns the old value which means you can do this without a perf hit in the common case.